### PR TITLE
info: print requirements

### LINKED ISSUE #4049: homebrew failing to operate correctly after update <10.13.3-x86_64>
- I ran `brew update` twice and can still reproduce the problem.
- I ran `brew doctor`, fixed all issues and can still reproduce the problem.
- I ran `brew config` and `brew doctor` and included their output below

To help us debug your issue please explain:
- I was attempting to upgrade hadoop, assuming there was a newer version.  There isn't.
- after submitting the brew upgrade hadoop command homebrew went into its normal updating homebrew process.  during this process it appears to have 
==> Renamed Formulae
php70 -> php@7.0
==> Deleted Formulae
arm
- at the end of the command execution it was showing the below error.  Now when i run brew services list or anything else it shows that error.

**Homebrew stopped working after/during the execution of my attempt to upgrade hadoop.**

$ brew upgrade hadoop
Updating Homebrew...
==> Auto-updated Homebrew!
Updated Homebrew from 1376b9e4 to 46821084.
Updated 6 taps (homebrew/core, homebrew/php, pivotal/tap, caskroom/fonts, caskroom/versions, caskroom/cask).
==> New Formulae
annie                    cquery                   jrtplib                  ssh-permit-a38           trezor-agent
apache-arrow-glib        gr-osmosdr               jsonrpc-glib             template-glib            util-linux
aws-es-proxy             hcloud                   libdazzle                tox
==> Updated Formulae
atk ✔                          convox                         grpc                           optipng
chromedriver ✔                 cryptopp                       gsoap                          osm2pgrouting
couchdb ✔                      darcs                          gst-plugins-good               osquery
erlang@19 ✔                    dartsim                        gtksourceview3                 packmol
gdk-pixbuf ✔                   davmail                        handbrake                      passenger
glib ✔                         dcos-cli                       haproxy                        pcb
gobject-introspection ✔        ddgr                           heroku                         pdal
harfbuzz ✔                     dependency-check               htslib                         pgcli
htop ✔                         di                             hugo                           pgroonga
jenkins ✔                      django-completion              internetarchive                picard-tools
nginx ✔                        docfx                          ipython                        pivotal/tap/gemfire
nifi ✔                         doctl                          ipython@5                      pivotal/tap/tcserver
node ✔                         dscanner                       ircd-hybrid                    plantuml
pango ✔                        duck                           jena                           postgis
pivotal/tap/springboot ✔       embulk                         jenkins-job-builder            pugixml
sbt ✔                          evince                         jhipster                       qpid-proton
zsh ✔                          exomizer                       json-glib                      raylib
abnfgen                        exploitdb                      kubectx                        robot-framework
agda                           eye-d3                         lbdb                           rocksdb
allegro                        faas-cli                       libcouchbase                   saltstack
angular-cli                    fail2ban                       libgit2                        samtools
apibuilder-cli                 ffmbc                          libgit2-glib                   singular
apktool                        firebase-cli                   libphonenumber                 skafos
armor                          flatbuffers                    librsvg                        solr
asciinema                      flawfinder                     libsoup                        sourcekitten
asio                           fluent-bit                     libuv                          sourcery
at-spi2-atk                    fn                             linkerd                        srt
at-spi2-core                   folly                          lldpd                          ssh-copy-id
aws-sdk-cpp                    fonttools                      lmod                           swiftlint
awscli                         fq                             lolcat                         syncthing
bcftools                       freetds                        maxwell                        talloc
bedops                         frege                          mediaconch                     teleport
bettercap                      frugal                         mednafen                       terraform
bitcoin                        gcab                           mercurial                      terragrunt
bitrise                        geckodriver                    metabase                       testssl
botan                          gedit                          miniupnpc                      tinyxml2
braid                          getmail                        mint                           todo-txt
byteman                        git                            mitmproxy                      tomcat
c14-cli                        git-annex                      mm-common                      uhd
calicoctl                      git-town                       mongo-c-driver                 unbound
certbot                        gitbucket                      nats-streaming-server          vala
chapel                         gitfs                          ncmpc                          vte3
checkstyle                     gitg                           nco                            webpack
chronograf                     gitless                        neofetch                       wine
clamav                         gjs                            notmuch                        wireshark
clinfo                         glade                          ntl                            xtensor
closure-compiler               gnome-builder                  nuget                          youtube-dl
cockroach                      google-benchmark               ocrmypdf                       zabbix
cocoapods                      grails                         odpi                           zenity
collectd                       graphene                       offlineimap
conan                          gron                           open-scene-graph
conjure-up                     groonga                        openssh
==> Renamed Formulae
php70 -> php@7.0
==> Deleted Formulae
arm
Error: nil value for strip
Please report this bug:
  https://docs.brew.sh/Troubleshooting
/usr/local/Homebrew/Library/Homebrew/patch.rb:21:in `create'
/usr/local/Homebrew/Library/Homebrew/software_spec.rb:202:in `patch'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2271:in `block in patch'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2271:in `each'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2271:in `patch'
/usr/local/opt/rstudio-server/.brew/rstudio-server.rb:116:in `<class:RstudioServer>'
/usr/local/opt/rstudio-server/.brew/rstudio-server.rb:1:in `load_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:26:in `module_eval'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:26:in `load_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:46:in `load_formula_from_path'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:101:in `load_file'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:92:in `klass'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:88:in `get_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:280:in `factory'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:314:in `rescue in from_keg'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:310:in `from_keg'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:294:in `from_rack'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1388:in `block in installed'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1386:in `each'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1386:in `flat_map'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1386:in `installed'
/usr/local/Homebrew/Library/Homebrew/cmd/update-report.rb:505:in `migrate_formula_rename'
/usr/local/Homebrew/Library/Homebrew/cmd/update-report.rb:100:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/update-report.rb:100:in `update_report'
/usr/local/Homebrew/Library/Homebrew/brew.rb:101:in `<main>'
Error: hadoop 3.0.0 already installed

~$ brew config
HOMEBREW_VERSION: 1.6.0-2-g4682108
ORIGIN: https://github.com/Homebrew/brew
HEAD: 468210840ca4a80b2f314736ddb9905181a8b3ae
Last commit: 12 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: a28b6d60078648c05a4955973051f678eca393b2
Core tap last commit: 2 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DEV_CMD_RUN: 1
CPU: octa-core 64-bit skylake
Homebrew Ruby: 2.3.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby
Clang: 9.0 build 900
Git: 2.6.4 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 9.0.4, 1.8.0_162, 1.8.0_151, 1.8.0_144, 1.8.0_131, 1.8.0_77
macOS: 10.13.3-x86_64
CLT: 9.2.0.0.1.1510905681
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11

~$ brew doctor
Error: nil value for strip
Please report this bug:
  https://docs.brew.sh/Troubleshooting
/usr/local/Homebrew/Library/Homebrew/patch.rb:21:in `create'
/usr/local/Homebrew/Library/Homebrew/software_spec.rb:202:in `patch'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2271:in `block in patch'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2271:in `each'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2271:in `patch'
/usr/local/opt/rstudio-server/.brew/rstudio-server.rb:116:in `<class:RstudioServer>'
/usr/local/opt/rstudio-server/.brew/rstudio-server.rb:1:in `load_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:26:in `module_eval'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:26:in `load_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:46:in `load_formula_from_path'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:101:in `load_file'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:92:in `klass'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:88:in `get_formula'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:280:in `factory'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:314:in `rescue in from_keg'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:310:in `from_keg'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:294:in `from_rack'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1388:in `block in installed'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1386:in `each'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1386:in `flat_map'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1386:in `installed'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:806:in `check_missing_deps'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:46:in `block in doctor'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:38:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:38:in `doctor'
/usr/local/Homebrew/Library/Homebrew/brew.rb:101:in `<main>'

### PR DESCRIPTION
**Pull request by ilovezfs**
Closes Homebrew/homebrew-core#4049.

### CODE REVIEW

File: Library/Homebrew/cmd/info.rb [hunk 1]
```diff
@@ -146,6 +146,22 @@ def info_formula(f)
...
+      %w[build required recommended optional].map do |type|
+        reqs = if type == "build"
+          f.requirements.select(&:build?).uniq
+        elsif type == "required"
+          f.requirements.select(&:required?).uniq
+        elsif type == "recommended"
+          f.requirements.select(&:recommended?).uniq
+        elsif type == "optional"
+          f.requirements.select(&:optional?).uniq
+        end
```
Comment (bfontaine) [inline Library/Homebrew/cmd/info.rb]: Can’t you do `case type` instead of `if`/`elsif`/etc?

Comment (jawshooah) [inline Library/Homebrew/cmd/info.rb]: For that matter, you could replace the entire block with `f.requirements.select(&:"#{type}?")`

**Comment by MikeMcQuaid** [comment]
@ilovezfs Why the close?

**Comment by ilovezfs** [comment]
lack of patience

**Comment by apjanke** [comment]
I have some patience lying around. Picking this up in #1004.

**Comment by ilovezfs** [comment]
Thanks Andrew

**Pull request closed without merging**